web-stories.php

```diff
@@ -108,19 +108,34 @@ function web_stories_get_compat_instance() {
 		require_once WEBSTORIES_PLUGIN_DIR_PATH . '/includes/compat/Web_Stories_Compatibility.php';
 	}
 
+	$js_assets_dir = WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/';
+
+	$required_files = array(
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/web-stories-embed-block.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/includes/vendor/autoload.php',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/third-party/vendor/scoper-autoload.php',
+	);
+
+	// Some scripts are chunked -- add all chunks to $required_files.
+	$chunked_scripts = array( 'edit-story', 'stories-dashboard' );
+	foreach ( $chunked_scripts as $script ) {
+		$script_file = $js_assets_dir . $script . '.js';
+		$chunks_file = $js_assets_dir . $script . '.chunks.php';
+
+		$chunks    = is_readable( $chunks_file ) ? require $chunks_file : [];
+		$js_chunks = isset( $chunks['js'] ) ? $chunks['js'] : [];
+
+		$required_files[] = $script_file;
+		foreach ( $js_chunks as $filename ) {
+			$required_files[] = $js_assets_dir . $filename . '.js';
+		}
+	}
```

Comment:
The goal of `$required_files` is to have a short list of files to check for to see whether the plugin was properly built using `npm install` and `composer install`. We're not interested in checking for all possible build files here, so we don't need to add those chunk files to the list here.

Reply:
Agreed, a short list is by design here. Check all these files on every page request is wasteful

Reply:
SGTM